wakuv2/waku_test.go

```diff
@@ -280,3 +282,103 @@ func TestWakuV2Filter(t *testing.T) {
 
 	require.NoError(t, w.Stop())
 }
+
+func TestWakuV2Store(t *testing.T) {
+	// Configuration for the first Waku node
+	sql1, err := helpers.SetupTestMemorySQLDB(appdatabase.DbInitializer{})
+	require.NoError(t, err)
+	config1 := &Config{
+		Port:              0,
+		EnableDiscV5:      false,
+		DiscoveryLimit:    20,
+		EnableStore:       true,
+		StoreCapacity:     100,
+		StoreSeconds:      3600,
+		KeepAliveInterval: 10,
+	}
+
+	// Start the first Waku node
+	w1, err := New("", "", config1, nil, sql1, nil, nil, nil)
+	require.NoError(t, err)
+	require.NoError(t, w1.Start())
+	defer func() {
+		require.NoError(t, w1.Stop())
+	}()
+
+	// Configuration for the second Waku node
+	sql2, err := helpers.SetupTestMemorySQLDB(appdatabase.DbInitializer{})
+	require.NoError(t, err)
+	config2 := &Config{
+		Port:              0,
+		EnableDiscV5:      false,
+		DiscoveryLimit:    20,
+		EnableStore:       true,
+		StoreCapacity:     100,
+		StoreSeconds:      3600,
+		KeepAliveInterval: 10,
+	}
+
+	// Start the second Waku node
+	w2, err := New("", "", config2, nil, sql2, nil, nil, nil)
+	require.NoError(t, err)
+	require.NoError(t, w2.Start())
+	defer func() {
+		require.NoError(t, w2.Stop())
+	}()
+
+	// Connect the two nodes directly
+	peer2Addr := w2.node.ListenAddresses()[0].String()
+	err = w1.node.DialPeer(context.Background(), peer2Addr)
+	require.NoError(t, err)
+
+	// Sanity check, not great, but it's probably helpful
+	options := func(b *backoff.ExponentialBackOff) {
+		b.MaxElapsedTime = 30 * time.Second
+	}
+	err = tt.RetryWithBackOff(func() error {
+		if len(w1.Peers()) == 0 {
+			return errors.New("no peers discovered")
+		}
+		return nil
+	}, options)
+	require.NoError(t, err)
+
+	// Wait for the nodes to discover each other
+	time.Sleep(1 * time.Second)
+
+	// Create a filter for the second node to catch messages
+	filter := &common.Filter{
+		Messages:      common.NewMemoryMessageStore(),
+		ContentTopics: common.NewTopicSetFromBytes([][]byte{{1, 2, 3, 4}}),
+	}
+
+	_, err = w2.Subscribe(filter)
+	require.NoError(t, err)
+
+	// Send a message from the first node
+	msgTimestamp := w1.CurrentTime().UnixNano()
+	contentTopic := maps.Keys(filter.ContentTopics)[0]
+	_, err = w1.Send(relay.DefaultWakuTopic, &pb.WakuMessage{
+		Payload:      []byte{1, 2, 3, 4, 5},
+		ContentTopic: contentTopic.ContentTopic(),
+		Version:      0,
+		Timestamp:    msgTimestamp,
+	})
+	require.NoError(t, err)
+
+	// Wait for the message to be transferred
+	time.Sleep(1 * time.Second)
```

Comment:
In a similar fashion to connection status updates, we have this for envelope events: https://github.com/status-im/status-go/blob/8e7d0a523e803f5f3b62f9b04deb3dd67449da16/wakuv2/waku.go#L1419
One can subscribe to these envelope events via `Waku.SubscribeEnvelopeEvents`

Reply:
So maybe this way we can obviate the need for `Sleep` statements.